webapp/src/main/java/com/box/l10n/mojito/service/ai/openai/OpenAILLMService.java

```diff
@@ -101,15 +103,46 @@ public AICheckResponse executeAIChecks(AICheckRequest aiCheckRequest) {
                     textUnit -> textUnit,
                     (existing, replacement) -> existing,
                     HashMap::new));
-    Map<String, List<AICheckResult>> results = new HashMap<>();
+
+    Map<String, CompletableFuture<List<AICheckResult>>> checkResultsBySource = new HashMap<>();
     textUnitsUniqueSource
         .values()
         .forEach(
             textUnit -> {
-              List<AICheckResult> aiCheckResults = checkString(textUnit, prompts, repository);
-              results.put(textUnit.getSource(), aiCheckResults);
+              CompletableFuture<List<AICheckResult>> checkResultFuture =
+                  CompletableFuture.supplyAsync(() -> checkString(textUnit, prompts, repository));
```

Comment:
Should we batch our way through the list here e.g send 10 strings for evaluation, as they come back send another 10 until we've worked our way through the list? Otherwise if we get a PR with a large number of strings we'll send all reqs at the same time and likely hit rate limits

Reply:
CompletableFuture docs: https://docs.oracle.com/javase/8/docs/api/index.html?java/util/concurrent/CompletableFuture.html state that the common ForkJoinPool is used so level of concurrency looks to set at the number of processors available